### PR TITLE
Add validation tests for *Bit builtin functions

### PR DESCRIPTION
**Pull request by lokokung**
Note that `insertBits` will come in a follow up because it has multiple arguments unlike these simpler ones.

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [x] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Comment by lokokung** [comment]
Note that these are also practically identical to the ones in https://github.com/gpuweb/cts/pull/3498 since these builtins all take the same argument types.

File: src/webgpu/shader/validation/expression/call/builtin/firstLeadingBit.spec.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+Validates that float arguments are rejected by ${builtin}()
+`
+  )
+  .params(u => u.combine('type', keysOf(kFloatTypes)))
+  .fn(t => {
+    const type = kFloatTypes[t.params.type];
+    validateConstOrOverrideBuiltinEval(
+      t,
+      builtin,
+      /* expectedResult */ false,
```
Comment (dj2) [inline src/webgpu/shader/validation/expression/call/builtin/firstLeadingBit.spec.ts:64]: Can you add a case into here that will pass, and then conditionalize this? We like to make sure the tests are failing due to the input not other issues.

File: src/webgpu/shader/validation/expression/call/builtin/firstLeadingBit.spec.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  good: '(1u)',
+  bad_no_parens: '',
+  // Bad number of args
+  bad_too_few: '()',
+  bad_too_many: '(1u,2u)',
+  // Bad value for arg 0 (Note that float type arguments are handled in 'float_argument' above)
+  bad_0bool: '(false)',
+  bad_0array: '(array(1u))',
+  bad_0struct: '(modf(2.2))',
+};
```
Comment (dj2) [inline src/webgpu/shader/validation/expression/call/builtin/firstLeadingBit.spec.ts:80]: Few more to add here, textures, samplers, matrices, vector, atomic

File: src/webgpu/shader/validation/expression/call/builtin/firstLeadingBit.spec.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    t.expectCompileResult(
+      t.params.arg === 'good',
+      `const c = ${builtin}${kArgCases[t.params.arg]};`
+    );
+  });
+
+g.test('must_use')
+  .desc(`Result of ${builtin} must be used`)
+  .fn(t => {
+    t.expectCompileResult(false, `fn f() { ${builtin}${kArgCases['good']}; }`);
```
Comment (dj2) [inline src/webgpu/shader/validation/expression/call/builtin/firstLeadingBit.spec.ts:95]: Can you conditionalize this and make a version which assigns to a phony to verify it passes?

**Review by dj2 — COMMENTED** [review]
Comments on the first test apply to the other two test files as well.

File: src/webgpu/shader/validation/expression/call/builtin/firstLeadingBit.spec.ts (hunk 2 above)
Comment (dneto0) [reply src/webgpu/shader/validation/expression/call/builtin/firstLeadingBit.spec.ts:80]: We haven't done those types in other similar tests.  So I'd be ok with filing a bug to followup to do this more broadly.

Also, textures and samplers can't be constructed in the middle of an expression, so we'd have to add a preamble to declare those variables.

File: src/webgpu/shader/validation/expression/call/builtin/firstLeadingBit.spec.ts (hunk 3 above)
Comment (dneto0) [reply src/webgpu/shader/validation/expression/call/builtin/firstLeadingBit.spec.ts:95]: Agreed, like we've done in others like `sin` and `cos`

File: src/webgpu/shader/validation/expression/call/builtin/firstLeadingBit.spec.ts (hunk 2 above)
Comment (dneto0) [reply src/webgpu/shader/validation/expression/call/builtin/firstLeadingBit.spec.ts:80]: Atomic is another case you can't construct in an expresison. you'd need a variable declaration somewhere.

**Review by dneto0 — CHANGES_REQUESTED** [review]
Please update the must_use tests to be like the other ones such as `cos`:

https://github.com/gpuweb/cts/blob/916580420fa306db53e7934c172b1a94f42aba16/src/webgpu/shader/validation/expression/call/builtin/cos.spec.ts#L108 

I'm ok adding a TODO for the other bad type cases that require a declared variable elsewhere: textures, samplers, atomics.

**Comment by dj2** [comment]
You can see an example of some of those tests in https://github.com/gpuweb/cts/blob/c06f419947827ece5335e4593ca466a31166c7d9/src/webgpu/shader/validation/expression/call/builtin/acos.spec.ts#L188

File: src/webgpu/shader/validation/expression/call/builtin/firstLeadingBit.spec.ts (hunk 2 above)
Comment (dj2) [reply src/webgpu/shader/validation/expression/call/builtin/firstLeadingBit.spec.ts:80]: These are the tests I've been adding as I worked down the list of builtins. You can see some examples in https://github.com/gpuweb/cts/blob/c06f419947827ece5335e4593ca466a31166c7d9/src/webgpu/shader/validation/expression/call/builtin/acos.spec.ts#L188

Comment (lokokung) [reply src/webgpu/shader/validation/expression/call/builtin/firstLeadingBit.spec.ts:158]: Done.

File: src/webgpu/shader/validation/expression/call/builtin/firstLeadingBit.spec.ts (hunk 3 above)
Comment (lokokung) [reply src/webgpu/shader/validation/expression/call/builtin/firstLeadingBit.spec.ts]: Done.

File: src/webgpu/shader/validation/expression/call/builtin/firstLeadingBit.spec.ts (hunk 1 above)
Comment (lokokung) [reply src/webgpu/shader/validation/expression/call/builtin/firstLeadingBit.spec.ts]: Done.

**Review by dneto0 — APPROVED** [review]
Thanks

**Pull request merged**